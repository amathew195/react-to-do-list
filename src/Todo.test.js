import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Todo from "./Todo";

const testTodo = {
  id: 1,
  title: 'test',
  priority: 1,
  description: 'this is a mock',
}

describe("renders", function () {
  it("renders without fail", function () {
    render(<Todo todo={testTodo}/>);
  });

  it("test that it renders with title, priority, description", function() {
    const { container, debug } = render(<Todo todo={testTodo}/>);

    expect(container).toContainHTML('test');
    expect(container).toContainHTML('(priority: 1)');
    expect(container).toContainHTML('this is a mock');

  });
});

describe("snapshots", function() {
  it("matches snapshot", function() {
      const { asFragment } = render(<Todo todo={testTodo}/>);
      expect(asFragment()).toMatchSnapshot();
  });
});
