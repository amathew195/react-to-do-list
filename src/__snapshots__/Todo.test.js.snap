// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <div>
      <b>
        test
      </b>
       
      <small>
        (priority: 1)
      </small>
    </div>
    <div>
      <small>
        this is a mock
      </small>
    </div>
  </div>
</DocumentFragment>
`;
